resources/aws_ec2.go

```diff
@@ -486,6 +489,68 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsEndpoint := obj.snsServer()
+	// shutdown the sns endpoint when we're done
```

Comment:
If the object of this is to run as this Watch method is shutting down, then I believe this whole method can be replace with a defer instead of this equally behaving, yet more heavyweight approach. Do you agree?

Reply:
This was my first approach. Hence the question about not using closeChan. If I don't need it, I think I should remove closeChan from the struct in a new commit before this one.

Reply:
No, you still need closeChan to shutdown the other goroutine :) Just fix the things in the comments, the rest is perfect!